packages/task/src/common/task-protocol.ts

```diff
@@ -101,9 +111,92 @@ export interface TaskExitedEvent {
     readonly config?: TaskConfiguration;
 }
 
+export interface TaskOutputEvent {
+    readonly taskId: number;
+    readonly ctx?: string;
+    readonly line: string;
+}
+
+export interface TaskOutputProcessedEvent {
+    readonly taskId: number;
+    readonly ctx?: string;
+    readonly problems?: ProblemMatch[];
+}
+
 export interface TaskClient {
     onTaskExit(event: TaskExitedEvent): void;
     onTaskCreated(event: TaskInfo): void;
     onDidStartTaskProcess(event: TaskInfo): void;
     onDidEndTaskProcess(event: TaskExitedEvent): void;
+    onDidProcessTaskOutput(event: TaskOutputProcessedEvent): void;
+}
+
+export interface TaskDefinition {
+    taskType: string;
+    properties: {
+        required: string[];
+        all: string[];
+    }
+}
+
+export interface WatchingMatcherContribution {
+    // If set to true the background monitor is in active mode when the task starts.
+    // This is equals of issuing a line that matches the beginPattern
+    activeOnStart?: boolean;
+    beginsPattern: string | WatchingPattern;
+    endsPattern: string | WatchingPattern;
+}
+
+export interface ProblemMatcherContribution {
+    name?: string;
+    label: string;
+    deprecated?: boolean;
+
+    owner: string;
+    source?: string;
+    applyTo?: string;
+    fileLocation?: 'absolute' | 'relative' | string[];
+    pattern: string | ProblemPatternContribution | ProblemPatternContribution[];
+    severity?: string;
+    watching?: WatchingMatcherContribution; // deprecated. Use `background`.
+    background?: WatchingMatcherContribution;
+}
+
+export interface ProblemPatternContribution {
+    name?: string;
+    regexp: string;
+
+    kind?: string;
+    file?: number;
+    message?: number;
+    location?: number;
+    line?: number;
+    character?: number;
+    endLine?: number;
+    endCharacter?: number;
+    code?: number;
+    severity?: number;
+    loop?: boolean;
+}
+
+export const TaskDefinitionRegistry = Symbol('TaskDefinitionRegistry');
```

Comment:
- Why do you need all these symbols and interfaces? Do you have an alternative implementations?
- Why does it belong to protocol? Is it supposed to work over JSON-RPC? 

If not please delete symbols and interrfaces and just use classes intead of them. Otherwise it is just boilerplate. Review code for such things.

Reply:
Added it to the coding guide: https://github.com/theia-ide/theia/wiki/Coding-Guidelines#classes-over-interfaces

Reply:
done.